src/main/java/teamdraco/finsandstails/common/entities/CrownedHorateeEntity.java

```diff
@@ -0,0 +1,661 @@
+		return p_30182_.getY() < p_30180_.getSeaLevel() + 4 && CrownedHorateeEntity.onSandOrGravel(p_30180_, p_30182_) && isBrightEnoughToSpawn(p_30180_, p_30182_);
+	}
+
+	public static boolean onSandOrGravel(BlockGetter p_57763_, BlockPos p_57764_) {
+		return isSandOrGravel(p_57763_, p_57764_.below());
+	}
+
+	public static boolean isSandOrGravel(BlockGetter p_57801_, BlockPos p_57802_) {
+		return p_57801_.getBlockState(p_57802_).is(BlockTags.SAND) || p_57801_.getBlockState(p_57802_).is(Tags.Blocks.GRAVEL);
+	}
+
+
+	public void travel(Vec3 p_27490_) {
+		if (this.isEffectiveAi() && this.isInWater() && !this.isOnGround()) {
+			this.moveRelative(0.05F, p_27490_);
+			this.move(MoverType.SELF, this.getDeltaMovement());
+			this.setDeltaMovement(this.getDeltaMovement().scale(0.9D));
+		} else {
+			super.travel(p_27490_);
+		}
+
+	}
+
+	@Override
+	protected PathNavigation createNavigation(Level worldIn) {
+		GroundPathNavigation groundPathNavigation = new GroundPathNavigation(this, level);
+		groundPathNavigation.setCanFloat(false);
+		return groundPathNavigation;
+	}
+
+	@Override
+	public boolean canBreatheUnderwater() {
+		return true;
+	}
+
+	@Override
+	protected void tickLeash() {
+		super.tickLeash();
+	}
+
+	protected void handleAirSupply(int p_149194_) {
+		if (this.isAlive() && !this.isInWaterRainOrBubble() && !this.isBaby()) {
+			this.setAirSupply(p_149194_ - 1);
+			if (this.getAirSupply() == -20) {
+				this.setAirSupply(0);
+				this.hurt(DamageSource.DRY_OUT, 2.0F);
+			}
+		} else {
+			this.setAirSupply(this.getMaxAirSupply());
+		}
+	}
+
+	public void rehydrate() {
+		int i = this.getAirSupply() + 1200;
+		this.setAirSupply(Math.min(i, this.getMaxAirSupply()));
+	}
+
+	public int getMaxAirSupply() {
+		return 6000;
+	}
+
+
+	@Override
+	public void registerControllers(AnimationData data) {
+		data.addAnimationController(new AnimationController<>(this, "controller", 5, this::predicrownedHorateee));
+		data.addAnimationController(new AnimationController<>(this, "miscController", 0, this::miscPredicate));
+	}
+
+	@Override
+	public AnimationFactory getFactory() {
+		return factory;
+	}
+
+	@Override
+	public boolean removeWhenFarAway(double p_21542_) {
+		return false;
+	}
+
+	@Nullable
+	@Override
+	protected SoundEvent getAmbientSound() {
+		return FTSounds.HORATEE_AMBIENT.get();
+	}
+
+	@Nullable
+	@Override
+	protected SoundEvent getHurtSound(DamageSource p_21239_) {
+		return FTSounds.HORATEE_HURT.get();
+	}
+
+	@Nullable
+	@Override
+	protected SoundEvent getDeathSound() {
+		return FTSounds.HORATEE_DEATH.get();
+	}
+
+	private boolean isMoving() {
+		return this.getDeltaMovement().lengthSqr() > 1.0E-7D;
+	}
+
+	private <E extends IAnimatable> PlayState miscPredicate(AnimationEvent<E> event) {
+		if (this.isBubbleCharge()) {
+			event.getController().setAnimation(new AnimationBuilder().addAnimation("animation.model.bubble", true));
+		} else {
+			event.getController().setAnimation(new AnimationBuilder());
```

Comment:
Why are you setting an empty builder, especially since you're then just returning stop? Just return stop.